lighthouse-core/gather/gatherers/dobetterweb/optimized-images.js

```diff
@@ -101,20 +103,50 @@ class OptimizedImages extends Gatherer {
    * @return {!Promise<{originalSize: number, jpegSize: number, webpSize: number}>}
    */
   calculateImageStats(driver, networkRecord) {
-    const param = JSON.stringify(networkRecord.url);
-    const script = `(${getOptimizedNumBytes.toString()})(${param})`;
-    return driver.evaluateAsync(script).then(stats => {
-      const isBase64DataUri = networkRecord.isBase64DataUri;
-      const base64Length = networkRecord.url.length - networkRecord.url.indexOf(',') - 1;
-      return {
-        originalSize: isBase64DataUri ? base64Length : networkRecord.resourceSize,
-        jpegSize: isBase64DataUri ? stats.jpeg.base64 : stats.jpeg.binary,
-        webpSize: isBase64DataUri ? stats.webp.base64 : stats.webp.binary,
-      };
+    let uriPromise = Promise.resolve(networkRecord.url);
+
+    // Get the request body for cross-origin images to circumvent canvas CORS protections
```

Comment:
*total* nit, but could start the comment with the conditional to make the context super clear: `// For cross-origin images, get the request body to circumvent canvas CORS protections`